conda_build/source.py

```diff
@@ -806,15 +803,18 @@ def try_apply_patch(patch, patch_args, cwd, stdout, stderr):
             check_call_env([git, 'am', '-3', '--committer-date-is-author-date', path],
                            cwd=src_dir, stdout=stdout, stderr=stderr, env=git_env)
             config.git_commits_since_tag += 1
+        elif patch_exe is None or len(patch_exe) == 0:
+            raise RuntimeError(
+                dedent(
+                    f"""
+                    Cannot use 'git' (not a git repo and/or patch) and did not find 'patch' in:
+                        {os.pathsep.join(external.dir_paths)}
+                    You can install 'patch' using apt-get, yum (Linux), Xcode (MacOSX),
+                    or conda, m2-patch (Windows),
+                    """
+                ).lstrip()
+            )
```

Comment:
Since `patch` is now a hard dependency do we remove this check entirely or do we move this check up with `patch_exe = external.find_executable("patch")` and error with something more abrasive?

Reply:
Yeah, that makes sense.